File: clean_api.py
import sys

with open('frontend/src/services/api.ts', 'r', encoding='utf-8') as f:
    lines = f.readlines()

out = []
i = 0
while i < len(lines):
    line = lines[i]
    if 'import {' in line and './mockFallback' in line:
        i += 1
        continue
    if 'if (forceDemoMode) {' in line:
        # skip this and next 3 lines assuming block is:
        # if (forceDemoMode) {
        #   notifyFallback();
        #   return ...
        # }
        while '}' not in lines[i]:
            i += 1
        i += 1
        continue
    
    if '} catch {' in line:
        out.append('  } catch (e) { throw e; }\n')
        # skip lines until the closing brace of the catch block
        i += 1
        while '}' not in lines[i]:
            i += 1
        i += 1
        continue
        
    out.append(line)

with open('frontend/src/services/api.ts', 'w', encoding='utf-8') as f:
    f.writelines(out)

File: clean_api.js
const fs = require('fs');
let lines = fs.readFileSync('frontend/src/services/api.ts', 'utf-8').split('\n');

const outLines = [];
let i = 0;

while (i < lines.length) {
    const line = lines[i];

    if (line.includes('import {') && line.includes('./mockFallback')) {
        // Skip import lines
        let j = i;
        while (j < lines.length && !lines[j].includes('} from \'./mockFallback\';')) {
            j++;
        }
        i = j + 1;
        continue;
    }

    if (line.includes('if (forceDemoMode) {')) {
        // Skip the if block
        let openBraces = 1;
        i++;
        while (i < lines.length && openBraces > 0) {
            if (lines[i].includes('{')) openBraces++;
            if (lines[i].includes('}')) openBraces--;
            i++;
        }
        continue;
    }

    if (line.includes('catch {')) {
        outLines.push(line.replace('catch {', 'catch (e) { throw e; }'));
        let openBraces = 1;
        i++;
        while (i < lines.length && openBraces > 0) {
            if (lines[i].includes('{')) openBraces++;
            if (lines[i].includes('}')) openBraces--;
            i++;
        }
        continue;
    }

    outLines.push(line);
    i++;
}

fs.writeFileSync('frontend/src/services/api.ts', outLines.join('\n'), 'utf-8');
console.log('Done cleaning api.ts');
